### DISCUSSION TITLE
does the staedi supports multiple ISA segments in an X12 EDI file

### DISCUSSION THREAD

**Discussion by dineshraghuramaiah** [xlate/staedi#352] [Q&A] [answered]
does the staedi supports multiple ISA segments in an X12 EDI file?

**Comment by MikeEdgar** [comment]
It should :-) If it doesn't it should be fixed

**Reply by dineshraghuramaiah** [reply]
Nope its not, below is the code.
while (reader.hasNext()) {
					event = reader.next();

					switch (event) {
					case START_INTERCHANGE:
						// Called at the beginning of the EDI stream once the X12 dialect is confirmed
						anStream = anStream.append("<Transaction-837>\n<Interchange>\n");
						break;
					case END_INTERCHANGE:
						// Called following the IEA segment
						anStream = anStream.append("</Interchange>\n</Transaction-837>\n");
						break;
}

**Reply by dineshraghuramaiah** [reply]
reader.hasNext() has ended after first occurrence of IEA

**Reply by MikeEdgar** [reply]
Thanks, I'll think about how to go about changing that.

**Reply by dineshraghuramaiah** [reply]
Thanks Mike

**Reply by MikeEdgar** [reply]
@dineshraghuramaiah this should be fixed now in version 1.23.0

**Reply by dineshraghuramaiah** [reply]
Thanks Mark, Let me try it.

**Reply by dineshraghuramaiah** [reply]
@MikeEdgar getting below error when i have second ISA in a EDI file. ErrorType= SEGMENT_EXCEEDS_MAXIMUM_USE,,Error= ISA at in segment ISA at position 236

**Reply by MikeEdgar** [reply]
@dineshraghuramaiah can you share a sample?

**Reply by MikeEdgar** [reply]
@dineshraghuramaiah , is the file you're processing somehow different than the new test case added with the multiple interchange support [1]? 

[1] https://github.com/xlate/staedi/blob/master/src/test/resources/x12/simple997-multiple-interchanges.edi

**Reply by MikeEdgar** [reply]
Thanks - it seems the existing test case is not detecting the same error. I'm looking in to a fix.

**Reply by MikeEdgar** [reply, marked as answer]
@dineshraghuramaiah - fixed in [v1.23.1](https://github.com/xlate/staedi/releases/tag/v1.23.1)
